game-core/src/main/java/games/strategy/net/IpFinder.java

```diff
@@ -79,26 +79,5 @@ public static InetAddress findInetAddress() throws SocketException, UnknownHostE
     }
     // all else fails, return localhost
     return InetAddress.getLocalHost();
-  } // end static findInetAddress()
-
-  private static boolean isPublicNetworkAddress(final InetAddress address) {
-    // stupid java signed byte type
-    final byte octet254 = (byte) 0xFE;
-    final byte[] bytes = address.getAddress();
-    // ip 4
-    if (bytes.length == 4) {
-      // http://en.wikipedia.org/wiki/Private_network
-      final byte octet169 = (byte) 0xA9;
-      final byte octet168 = (byte) 0xA8;
-      final byte octet172 = (byte) 0xAC;
-      final byte octet192 = (byte) 0xC0;
-      return (bytes[0] != 10) && (bytes[0] != octet172 || bytes[1] < 16 || bytes[1] > 31)
-          && (bytes[0] != octet192 || bytes[1] != octet168) && (bytes[0] != octet169 || bytes[1] != octet254);
-    }
-    // ip 6
-    // http://en.wikipedia.org/wiki/IPv6#Addressing
-    final byte octet252 = (byte) 0xFC;
-    return (bytes[0] != octet252 || bytes[1] != 0) && bytes[0] != octet254;
```

Comment:
This implementation doesn't exactly match what's in `Inet6Address#isSiteLocalAddress()` and `Inet6Address#isLinkLocalAddress()`.  I'm willing to believe this implementation is wrong, and the JDK implementation is correct, but I'm not sure.

Reply:
I'll trust the JDK ^^